opt out code added #14

rluetke opened the pull request:
Hi John,

I added  an optOut-function according to General Data Protection Regulation in Europe and https://developers.google.com/analytics/devguides/collection/gajs/#disable. 
But I did not add changes to blueprint / admin panel!
It works fine on my website https://blog.dmr-solutions.com/.

Regards, Ralf

rluetke committed:
opt out code added

escopecz commented:
Looks good! I'll have to find some time to test it, but if someone else would have a minute to do that please write your test results in a comment and I'll merge it if successful.

rluetke committed:
reaadme more detailed for opt out

rluetke commented:
Hi John,

meanwhile I commited the readme again with more details about using opt out.

And I did the following tests with my site https://blog.dmr-solutions.com/ 

- check JavaScript-Code in HTML source with and without "opt out" enabled: OK
- open Google Analytics realtime report for this site
- clicked some pages and check if this is listed in Google Analytics: OK
- list cookies and look for "ga-disable-UA-9493622-1" (with my tracking ID): not found = OK
- click on the opt-out-link: opt out message is shown = OK
- list cookies and look for "ga-disable-UA-9493622-1": found and is "true" = OK
- clicked some pages and check if this is listed in Google Analytics: not listed = OK
- delete cookie "ga-disable-UA-9493622-1"
- clicked some pages and check if this is listed in Google Analytics: OK

The opt out cookie is set and Google Analytics stopps tracking.
This worked as designed, but is only tested by me, the developer :-;

So, perhaps someone else may repeat the tests.

Regards, Ralf

rluetke commented:
Hi John,

but you still have to add the two new parameters in blueprints.yaml and language files and test it in admin plugin.

Regards, Ralf

escopecz commented:
@rluetke could you make all changes necessary for this feature, please?

rluetke committed:
opt out in admin panel

rluetke commented:
Hi John,

it took a while but now all changes for this feature are pushed to the repo. "Tracking Opt Out" can be configured in admin panel, too.

You could insert in the change.log e.g.:

+ new: Google Analytics Out Out
+ improved: spanish added for Admin Panel 

I did not make any changes in the change.log, only in the read.me.

Regards and happy new year, Ralf

rluetke committed:
opt out in admin panel as own tab

rluetke commented:
Hi John,

I moved the "opt out" parameters in the admin panel under an own tab and added some explanations there. I hope this makes "opt out" a bit clearer.

Regards, Ralf

yann1cks commented:
Hey Ralf, 

thank you very much for changes, works good for me! 

Could you also add a do-not-track option? I tried to add it but it does not work. Thats the way I tried it, shouldnt be to complicated:
https://www.paulfurley.com/google-analytics-do-not-track/

Regards
Yannick

rluetke commented:
Hi Yannick,

which Browser did you use?
See https://dev.to/corbindavenport/how-to-correctly-check-for-do-not-track-with-javascript-135d for cross browser functionality.

Regards, Ralf


Von: yann1cks [mailto:notifications@github.com]
Gesendet: Freitag, 17. Mai 2019 01:20
An: escopecz/grav-ganalytics
Cc: Ralf Lütke; Mention
Betreff: Re: [escopecz/grav-ganalytics] opt out code added #14 (#20)


Hey Ralf,

thank you very much for changes, works goog for me!

Could you also add a do-not-track option? I tried to add it but it does not work. Thats the way I tried it, shouldnt be to complicated:
https://www.paulfurley.com/google-analytics-do-not-track/

Regards
Yannick

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/escopecz/grav-ganalytics/pull/20?email_source=notifications&email_token=ACPXJ7YBWVBXCOVU4SVFOF3PVXTYHA5CNFSM4GJYNYL2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODVTJT4Q#issuecomment-493263346>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ACPXJ75TBBMMXI5CANVDPXTPVXTYHANCNFSM4GJYNYLQ>.
[ { "@context": "http://schema.org", "@type": "EmailMessage", "potentialAction": { "@type": "ViewAction", "target": "https://github.com/escopecz/grav-ganalytics/pull/20?email_source=notifications\u0026email_token=ACPXJ7YBWVBXCOVU4SVFOF3PVXTYHA5CNFSM4GJYNYL2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODVTJT4Q#issuecomment-493263346", "url": "https://github.com/escopecz/grav-ganalytics/pull/20?email_source=notifications\u0026email_token=ACPXJ7YBWVBXCOVU4SVFOF3PVXTYHA5CNFSM4GJYNYL2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODVTJT4Q#issuecomment-493263346", "name": "View Pull Request" }, "description": "View this Pull Request on GitHub", "publisher": { "@type": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

rluetke commented:
Hi John,

thanks!

Europe’s top court says active consent is needed for tracking cookies.<https://developereconomics.us1.list-manage.com/track/click?u=f5cdd9d9e59e9c39c83d7b50f&id=fc7c7cc252&e=7edce35a60> Europe’s top court has ruled that pre-checked consent boxes for dropping cookies are not legally valid. Consent must be obtained prior to storing or accessing non-essential cookies, such as tracking cookies for targeted advertising. Consent cannot be implied or assumed. Websites could be risking a big fine under EU privacy laws if they don’t obtain valid consent for tracking.

This means that tracking and cookie consent must work together.
I have to think a bit about the consequences for this plugin.

Regards, Ralf

Von: John Linhart [mailto:notifications@github.com]
Gesendet: Sonntag, 22. September 2019 14:00
An: escopecz/grav-ganalytics
Cc: Ralf Lütke; Mention
Betreff: Re: [escopecz/grav-ganalytics] opt out code added #14 (#20)


Merged #20<https://github.com/escopecz/grav-ganalytics/pull/20> into master.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/escopecz/grav-ganalytics/pull/20?email_source=notifications&email_token=ACPXJ76YLFY2PD56HX7PVUDQK5M4FA5CNFSM4GJYNYL2YY3PNVWWK3TUL52HS4DFWZEXG43VMVCXMZLOORHG65DJMZUWGYLUNFXW5KTDN5WW2ZLOORPWSZGOTYN5YBA#event-2652625924>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ACPXJ7ZIKMGPATRZRYLOSL3QK5M4FANCNFSM4GJYNYLQ>.
[ { "@context": "http://schema.org", "@type": "EmailMessage", "potentialAction": { "@type": "ViewAction", "target": "https://github.com/escopecz/grav-ganalytics/pull/20?email_source=notifications\u0026email_token=ACPXJ76YLFY2PD56HX7PVUDQK5M4FA5CNFSM4GJYNYL2YY3PNVWWK3TUL52HS4DFWZEXG43VMVCXMZLOORHG65DJMZUWGYLUNFXW5KTDN5WW2ZLOORPWSZGOTYN5YBA#event-2652625924", "url": "https://github.com/escopecz/grav-ganalytics/pull/20?email_source=notifications\u0026email_token=ACPXJ76YLFY2PD56HX7PVUDQK5M4FA5CNFSM4GJYNYL2YY3PNVWWK3TUL52HS4DFWZEXG43VMVCXMZLOORHG65DJMZUWGYLUNFXW5KTDN5WW2ZLOORPWSZGOTYN5YBA#event-2652625924", "name": "View Pull Request" }, "description": "View this Pull Request on GitHub", "publisher": { "@type": "Organization", "name": "GitHub", "url": "https://github.com" } } ]
